### ISSUE TITLE
Add a treatment for consistent appearance of content on focus

### ISSUE THREAD

**Issue by ericwbailey** [primer/design#338]
This issue arose from https://github.com/primer/react/pull/2593. 

The SplitPageLayout component uses a small form with an input and submit button to allow for adjusting the pane width without the need for pointer input. This form is visually hidden, meaning that in its current implementation that only screen reader users will be aware of its presence. 

Controls such as this [need to also be visually apparent on focus](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html). While only the SplitPageLayout component has this need right now, I anticipate other components needing similar functionality—and therefore a consistent visual treatment—as our Primer accessibility work continues.

Currently there is no one set way to handle this kind of content. The closest examples I am aware of are our Skip to main content link and our dashboard favorites reordering functionality:

<img width="455" alt="A blue rectangle containing the phrase, 'Skip to content' appended to the top lefthand corner of the viewport on GitHub's homepage." src="https://user-images.githubusercontent.com/634191/207390726-bcbdc0df-58a1-453b-8010-d073bb0f6434.png">

![The favorite repo sidebar on GitHub's dashboard. A button containing an arrow with an upward-facing chevron and a focus ring is placed next to a favorited repo, indicating that it may be moved up in the list of favorite repos.](https://user-images.githubusercontent.com/634191/207390972-152567ca-8931-476f-823d-288aea5ef870.png)

I am thinking that this is worth codifying as something that can be used as a subcomponent for other components, to help ensure a consistent, accessible experience that feels like it is a part of Primer. We'll likely want something along the lines of how [our Dialog component](https://primer.style/view-components/components/alpha/dialog) looks.

## Additional context

- [Slack thread in #accessibility-internal](https://github.slack.com/archives/C01UWR2MEUB/p1670430365991479)
- [Slack thread in #primer-design](https://github.slack.com/archives/CPA865D9S/p1670522653840849)
- [AppFrame work](https://primer.style/css/storybook/?path=%2Fstory%2Fcomponents-layout-appframe--playground)

**Comment by alliethu** [comment]
Closing in place of https://github.com/github/primer/issues/1591
